x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/group_title_renderers.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiSkeletonText, EuiTitle } from '@elastic/eui';
+import { isArray } from 'lodash/fp';
+import React, { memo } from 'react';
+import type { GroupPanelRenderer } from '@kbn/grouping/src';
+import { CardIcon } from '@kbn/fleet-plugin/public';
+import { useGetIntegrationFromRuleId } from '../../../hooks/alert_summary/use_get_integration_from_rule_id';
+import { GroupWithIconContent, RuleNameGroupContent } from '../../alerts_table/grouping_settings';
+import type { AlertsGroupingAggregation } from '../../alerts_table/grouping_settings/types';
+import { firstNonNullValue } from '../../../../../common/endpoint/models/ecs_safety_helpers';
+
+/**
+ * Returns renderers to be used in the `buttonContent` property of the EuiAccordion component used within the kbn-grouping package.
+ * It handles custom renders for the following fields:
+ * - signal.rule.id
+ * - kibana.alert.rule.name
+ * - host.name
+ * - user.name
+ * - source.ip
+ * For all the other fields the default renderer managed within the kbn-grouping package will be used.
+ *
+ * These go hand in hand with groupingOptions, groupStatsRenderer and groupStatsAggregations.
+ */
+export const groupTitleRenderers: GroupPanelRenderer<AlertsGroupingAggregation> = (
+  selectedGroup,
+  bucket,
+  nullGroupMessage
+) => {
+  switch (selectedGroup) {
+    case 'signal.rule.id':
+      return <IntegrationNameGroupContent title={bucket.key} />;
+    case 'kibana.alert.rule.name':
+      return isArray(bucket.key) ? (
+        <RuleNameGroupContent
+          ruleName={bucket.key[0]}
+          ruleDescription={
+            firstNonNullValue(firstNonNullValue(bucket.description?.buckets)?.key) ?? ''
+          }
+          tags={bucket.ruleTags?.buckets}
+        />
+      ) : undefined;
+    case 'host.name':
+      return (
+        <GroupWithIconContent
+          title={bucket.key}
+          icon="storage"
+          nullGroupMessage={nullGroupMessage}
+          dataTestSubj="host-name"
```

Comment:
Should these be prefixed with `alert-summary:grouping:host-name` or similar? Same for the others below

Reply:
why would this be needed?